packages/mdl-select/README.md

```diff
@@ -0,0 +1,270 @@
+into their custom components. Note that due to the nature of MDL Select, the adapter is quite
+complex. We try to provide as much guidance as possible, but we encourage developers to reach out
+to use via GH Issues or on Gitter if they run into problems.
+
+### Notes for component implementors
+
+The `MDLSelectFoundation` expects that the select component _controls an instance of
+`MDLSimpleMenu`_. We achieve this via composition in our vanilla `MDLSelect` component, and
+recommend a similar approach for framework authors.
+
+`MDLSelectFoundation` also has the ability to resize itself whenever its options change, via the
+`resize()` method. We recommend calling this method on initialization, or when the menu items are
+modified. For example, if building a react component, it may be appropriate to call `resize()`
+within `componentDidUpdate`.
+
+### Adapter API
+
+| Method Signature | Description |
+| --- | --- |
+| `addClass(className: string) => void` | Adds a class to the root element. |
+| `removeClass(className: string) => void` | Removes a class from the root element. |
+| `setAttr(attr: string, value: string) => void` | Sets attribute `name` to value `value` on the root element. |
+| `rmAttr(attr: string) => void` | Removes attribute `attr` from the root element. |
+| `registerInteractionHandler(type: string, handler: EventListener) => void` | Adds an event listener `handler` for event type `type` on the root element. |
+| `deregisterInteractionHandler(type: string, handler: EventListener) => void` | Removes an event listener `handler` for event type `type` on the root element. |
+| `focus() => void` | Focuses the root element |
+| `makeFocusable() => void` | Allows the root element to be focusable via keyboard. We achieve this by setting the root element's `tabIndex` property to `0`. |
+| `makeUnfocusable() => void` | Disallows the root element to be focusable via keyboard. We achieve this by setting the root element's `tabIndex` property to `-1`. |
+| `getComputedStyleValue(propertyName: string) => string` | Get the root element's computed style value of the given dasherized css property `propertyName`. We achieve this via `getComputedStyle(...).getPropertyValue(propertyName). `|
+| `setStyle(propertyName: string, value: string) => void` | Sets a dasherized css property `propertyName` to the value `value` on the root element. We achieve this via `root.style.setProperty(propertyName, value)`. |
+| `create2dRenderingContext() => {font: string, measureText: (string) => {width: number}}` | Returns an object which has the shape of a CanvasRenderingContext2d instance. Namely, it has a string property `font` which is writable, and a method `measureText` which given a string of text returns an object containing a `width` property specifying how wide that text should be rendered in the `font` specified by the font property. An easy way to achieve this is simply `document.createElement('canvas').getContext('2d');`. |
```

Comment:
I'd rather keep it at the foundation level, since it is an essential part of good UX for a select. I'd be hesitant to make a higher-level `computeWidth` or similar adapter method since that might be too _high-level_, even if we were to factor it into a util. Ideally, each adapter method is self-describing and can be written in a few lines or less of code